apps/api/src/index.ts

```diff
@@ -275,26 +275,28 @@ const mainServer = app.listen(PORT, "0.0.0.0", () => {
 // Start dedicated metrics server on separate port
 const metricsServer = startMetricsServer();
 
+let shuttingDown = false;
 // Graceful shutdown handler
-const gracefulShutdown = (signal: string) => {
+const gracefulShutdown = async (signal: string) => {
+  if (shuttingDown) return;
+  shuttingDown = true;
+
   console.log(
     `\n[${signal}] Received shutdown signal, closing servers gracefully...`,
   );
 
   // Close both servers
   mainServer.close(() => {
-    console.log("Main server closed");
-  });
+    console.log("[Shutdown] Main server closed");
```

Comment:
This function is **not** async, and neither is the execution of these callbacks.  This is a bit of a stylistic choice so we can definitely change if folks prefer that existing way.